### PR TITLE
[Datafactory] [Do not merge] Add upgrade api

### PR DESCRIPTION
**Pull request by leonliang33**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ x ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ x ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ x ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by hvermis — APPROVED** [review]

**Comment by anuchandy** [comment]
Requesting ARM review since this introduces a new api to stable API Version? 
cc @KrisBash @ravbhatnagar

**Comment by anuchandy** [comment]
@KrisBash @ravbhatnagar could you take a look please?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 1]
```diff
@@ -443,6 +443,48 @@
...
+          {
+            "$ref": "#/parameters/api-version"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/Factory"
+            }
```
Comment (sanjaiganesh) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: I don't think you would like to return the resource representation here.. just 200 (OK) should be good.

Comment (sanjaiganesh) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: + @ravbhatnagar

Comment (sanjaiganesh) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: Can you add more details on the upgrade behavior ? whether it is a one time thing, in the life cycle of a resource ? 
POST can't be expressed in ARM template and is discouraged for things that customer would like to express them in declarative manner (in template). 

---
In reply to: [264448182](https://github.com/Azure/azure-rest-api-specs/pull/5297#discussion_r264448182) [](ancestors = 264448182)

**Review by hvermis — CHANGES_REQUESTED** [review]
Please do not merge these changes. It seems we do not need this API.

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5403 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5403/commits/f83f3730df3ca5c613e49c9f8366428b90405827

**Comment by anuchandy** [comment]
Requesting for reviewer reassign since I'm OOF

**Comment by salameer** [comment]
@hvermis if this is not deleted please close this PR

**Pull request closed without merging**